package/MDAnalysis/analysis/base.py

```diff
@@ -430,25 +471,241 @@ def run(self, start=None, stop=None, step=None, frames=None,
+        scheduler : str, optional 
+            Enables parallel execution of :meth:`AnalysisBase.run`.
+            Available schedulers
+              - 'dask' implies execution using `dask` processes (need to install MDAnalysis[dask] for that to work)
+              - 'multiprocessing' works with standard python multiprocessing module
+              - `None` if you want to run things locally
+        n_workers : int, optional
+            number of workers. 
+            Will become number of processes in multiprocessing.Pool(...) with `multiprocessing` scheduler,
+            and `n_workers` with `dask` scheduler.
+
+        .. versionchanged:: 2.2.0
+            Added ability to analyze arbitrary frames by passing a list of
+            frame indices in the `frames` keyword argument.
+
+        .. versionchanged:: 2.5.0
+            Add `progressbar_kwargs` parameter, 
+            allowing to modify description, position etc of tqdm progressbars
+        """
+        if progressbar_kwargs and scheduler:
+            raise NotImplementedError("Can not display progress if scheduler is not None")
+
+        self._setup_frames(self._trajectory, start, stop, step, frames)
+        self._prepare()
+        self._setup_scheduler(scheduler=scheduler, n_workers=n_workers)
+        self._setup_bslices(start=start, stop=stop, step=step, frames=frames)
+
+
+        if scheduler is None: # fallback to the local scheduler
+            self._compute(bslice_idx=0, verbose=verbose, progressbar_kwargs=progressbar_kwargs)
+        else:
+            if scheduler == 'multiprocessing':
+                import multiprocessing
+
+                delayed = localdelayed
+                computations = [
+                    (self, bslice_idx)
+                    for bslice_idx in range(len(self._bslices))]
+
+                with multiprocessing.Pool(processes=self._scheduler_kwargs['n_workers']) as pool:
+                    results = pool.starmap(_self_compute_helper, computations)
+
+            elif scheduler == 'dask':
+                try:
+                    from dask.delayed import delayed
+                except ImportError:
+                    raise ImportError(
+                        "Please install dask to execute run() in parallel using scheduler='dask'")
+
+                computations = delayed(
+                    [
+                        delayed(self._compute)(bslice_idx=bslice_idx)
+                        for bslice_idx in range(len(self._bslices))
+                    ]
+                )
+                results = computations.compute(**self._scheduler_kwargs)
+
+            else:
+                raise ValueError("scheduler can be 'dask', 'multiprocessing' or None")
+
+            self._remote_results = results
+            self._parallel_conclude()
+
+            self.frames = np.array([obj.frames for obj in self._remote_results]).sum(axis=0)
+            self.times = np.array([obj.times for obj in self._remote_results]).sum(axis=0)
```

Comment:
>What happens here? Is the sum() an array concatenation??
no, but since both arrays were initialized as zeros, it effectively merges all non-zero places

>You might have to explain somewhere what the data look like that the workers have. 
You mean, in the documentation?

>Eg are all arrays the full size and workers only fill in their parts?
this one, yes.